openshift/templates/backup/backup-deploy.yaml

```diff
@@ -0,0 +1,432 @@
+      originate.
+    required: false
+    value: ""
+  - name: ENVIRONMENT_NAME
+    displayName: Environment Name (Environment Id)
+    description:
+      The name or Id of the environment.  This variable is used by the webhook
+      integration to identify the environment in which the backup notifications originate.
+    required: false
+    value: ""
+  - name: BACKUP_DIR
+    displayName: The root backup directory
+    description:
+      The name of the root backup directory.  The backup volume will be mounted
+      to this directory.
+    required: true
+    value: "/backups/"
+  - name: NUM_BACKUPS
+    displayName: The number of backup files to be retained
+    description:
+      Used for backward compatibility only.  Ignored when using the recommended
+      `rolling` backup strategy.  The number of backup files to be retained.  Used for
+      the `daily` backup strategy.
+    required: false
+    value: ""
+  - name: DAILY_BACKUPS
+    displayName: Number of Daily Backups to Retain
+    description:
+      The number of daily backup files to be retained.  Used for the `rolling`
+      backup strategy.
+    required: false
+    value: ""
+  - name: WEEKLY_BACKUPS
+    displayName: Number of Weekly Backups to Retain
+    description:
+      The number of weekly backup files to be retained.  Used for the `rolling`
+      backup strategy.
+    required: false
+    value: ""
+  - name: MONTHLY_BACKUPS
+    displayName: Number of Monthly Backups to Retain
+    description:
+      The number of monthly backup files to be retained.  Used for the `rolling`
+      backup strategy.
+    required: false
+    value: ""
+  - name: BACKUP_PERIOD
+    displayName: Period (d,m,s) between backups in a format used by the sleep command
+    description:
+      Used for backward compatibility only.  Ignored when using the recommended
+      `backup.conf` and cron backup strategy.  Period (d,m,s) between backups in a format
+      used by the sleep command
+    required: false
+    value: ""
+  - name: CONFIG_FILE_NAME
+    displayName: Config File Name
+    description: The name of the configuration file.
+    required: true
+    value: backup.conf
+  - name: CONFIG_MAP_NAME
+    displayName: Config Map Name
+    description: The name of the configuration map.
+    required: true
+    value: backup-conf
+  - name: CONFIG_MOUNT_PATH
+    displayName: Config Mount Path
+    description: The path to use to mount the config file.
+    required: true
+    value: "/"
+  - name: BACKUP_VOLUME_NAME
+    displayName: Backup Volume Name
+    description:
+      The name of the persistent volume used to store the backups.  Please
+      note, when using the recommended nfs-backup storage class the name of the pvc
+      MUST be taken from the manually provisioned claim; nfs-backup storage MUST be
+      provisioned manually.
+    required: true
+    value: bk-jcxjin-dev-r9c53stkggh6
```

Comment:
Ideally I'd like to see this value removed so - in combination with being a required field - forces the user to provide a value for it...

I suspect if someone runs this template and forgets to pass `BACKUP_VOLUME_NAME`, that **bk-jcxjin-dev-r9c53stkggh6** won't be there on their environment to work

Reply:
fair, I'll remove this.